docs/Version-support-policy.md

```diff
@@ -0,0 +1,32 @@
+[Back to overview](./README.md)
+
+# Version support policy
+
+As outlined in the [plugin announcement post](https://make.wordpress.org/core/2022/03/07/the-performance-lab-plugin-has-been-released/), the Performance Lab plugin is a collection of performance-related "feature projects" for WordPress core. These feature projects are represented in this plugin as individual standalone [modules](./Writing-a-module.md).
+
+All modules bundled in the Performance Lab plugin are at different stages of development and may be merged into WordPress core at different points. This document describes the version support policy for the different modules, with a special focus on backward compatibility.
+
+## WordPress core versions
+
+**The Performance Lab plugin commits to supporting the latest stable version of WordPress core.** With that said, the minimum WordPress version requirement will not be bumped unnecessarily, so realistically the plugin will usually support the latest _two_ stable versions of WordPress core. If or once a feature that is only available in the latest WordPress core stable release is required for a plugin module, the minimum requirement of the plugin will be bumped to that version.
+
+Supporting a greater array of older WordPress core versions would be detrimental for the plugin, as it is primarily intended for beta testing and often requires using features that have only been introduced in a recent WordPress core version.
+
+It needs to be clarified though that **the Performance Lab plugin will never require an alpha or beta version of a future WordPress core release**. This policy ensures that sites using the latest stable version of WordPress will always be able to use the plugin as well.
+
+## Removal of modules
+
+Once a module has been merged into WordPress core, there are two potential options for how to proceed with the module:
```

Comment:
I think we should standardize the way the modules are enabled currently is based on the settings but I think each module might be different so we should define a filterable function available in all modules like `function webp_uploads_should_module_load` with the specific logic for that module.

Reply:
This is a great point, we eventually need some way to not load a module in case the site is on a WordPress core version that already includes this feature (e.g. after a merge), and potentially we may even want to inform the admin about this when they access the plugin's settings screen.

That's unrelated to this documentation though, we can open a separate issue for it.

Reply:
:100: agree with that.